shuffle.sh

```diff
@@ -0,0 +1,109 @@
+
+# Clone everything
+cd $WORK_DIR
+git clone $CABLE_REPO $CABLE_DIR
+git clone $UM7_REPO $UM7_DIR
+
+# Check out the CABLE branch for the shuffle
+cd $CABLE_DIR
+git fetch origin
+git checkout $CABLE_BRANCH
+
+# Check out specific branches
+cd $UM7_DIR
+git fetch origin
+git checkout $UM7_BRANCH
+
+# Create a src/shared dir in CABLE
+mkdir -p $CABLE_SHARED_DIR
+
+# Move from offline to shared
+mv $CABLE_DIR/src/offline/cable_LUC_EXPT.F90 $CABLE_SHARED_DIR/
+mv $CABLE_DIR/src/offline/cable_phenology.F90 $CABLE_SHARED_DIR/
+mv $CABLE_DIR/src/offline/casa_ncdf.F90 $CABLE_SHARED_DIR/
+mv $CABLE_DIR/src/offline/casa_offline_inout.F90 $CABLE_SHARED_DIR/
+mv $UM7_ESM15_DIR/cable_pft_params_mod.F90 $CABLE_SHARED_DIR/
+mv $UM7_ESM15_DIR/cable_soil_params_mod.F90 $CABLE_SHARED_DIR/
+
+rm $CABLE_DIR/src/offline/cable_pft_params.F90
+rm $CABLE_DIR/src/offline/cable_soil_params.F90
+
+# Remove them from offline (done in move above)
+
+# Remove from ESM1.5 (none)
+
+# Remove from esm16
+rm -rf $CABLE_ESM16_DIR/cable_LUC_EXPT.F90
+rm -rf $CABLE_ESM16_DIR/cable_phenology.F90
+rm -rf $CABLE_ESM16_DIR/casa_ncdf.F90
+rm -rf $CABLE_ESM16_DIR/casa_offline_inout.F90
+
+# Add shared to CMakeLists.txt (manual)
+
+# Create the land_surface/cable dir
+mkdir -p $UM7_CABLE_DIR
+
+# Move source into it (not sure if we need to delete the src location?)
+cp $UM7_ESM15_DIR/casa_types.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/pack_mod_cbl.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_rad_driver.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_um_init.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_um_init_subrs.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_um_tech.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/allocate_soil_params_cbl.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/allocate_veg_params_cbl.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_cbm.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_explicit_driver.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_hyd_driver.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_implicit_driver.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cbl_um_init_soil.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cbl_um_init_soilsnow.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cbl_um_update_soilsnow.F90 $UM7_CABLE_DIR/
+cp $UM7_ESM15_DIR/cable_read_nml.F90 $UM7_CABLE_DIR/
+
+# casa_um_inout.F90 (currently in esm16/ but it needs to be deleted from there)
+# it is not there?
+
+# Put in $CABLE_DIR/src/coupled/esm/, copied from $UM7_DIR/src/atmosphere/CABLE/src/coupled/ESM1.5/
+cp $UM7_ESM15_DIR/cable_surface_types.F90 $CABLE_ESM_DIR/
+cp $UM7_ESM15_DIR/cable_landuse.F90 $CABLE_ESM_DIR/
+cp $UM7_ESM15_DIR/cable_define_types.F90 $CABLE_ESM_DIR/
+cp $UM7_ESM15_DIR/cable_iovars.F90 $CABLE_ESM_DIR/
+
+# Remove unnecessary directories
+rm -r $CABLE_ESM16_DIR
+rm -r $CABLE_ESM15_DIR
+rm -r $CABLE_DIR/src/coupled/ACCESS-CM2
+
+# Replace CABLE directory in UM7 with the one with the reorganised files
+rm -r $UM7_DIR/umbase_hg3/src/atmosphere/CABLE/src/
+cp -r $CABLE_DIR/src $UM7_DIR/umbase_hg3/src/atmosphere/
```

Comment:
cp -r $CABLE_DIR/src/* $UM7_DIR/umbase_hg3/src/atmosphere/CABLE/src
```suggestion
mkdir -p $UM7_DIR/umbase_hg3/src/atmosphere/CABLE/src
cp -r $CABLE_DIR/src/* $UM7_DIR/umbase_hg3/src/atmosphere/CABLE/src/
```

Reply:
@ccarouge, is this what you were thinking?

Reply:
I dont think it will let you mk 2 dirs at once. Do you already have .../CABLE/?

Reply:
> I dont think it will let you mk 2 dirs at once. Do you already have .../CABLE/?

From `man mkdir`

```
-p             Create intermediate directories as required.
```

I do it all the time, as long as you know the final leaf dir at the end, UNIX will handle the rest.

Reply:
Yes it is. But we might have an issue making it automatically: the codes in CABLE and UM7 repos are not completely sync... Should we leave this part to be done manually?